File: Uber/3/3.py
def peakTime(start, end):
	n = len(start)
	if(n == 0):
		return -1
	lastExitTime = 0
	for i in range(0, n):
		lastExitTime = max(lastExitTime, start[i])
		lastExitTime = max(lastExitTime, end[i])
	vehicleCount = (lastExitTime + 2) * [0]
	current = 0
	index = 0
	for i in range(0, n):
		vehicleCount[start[i]] += 1
		vehicleCount[end[i] + 1] -= 1
	maxVehicleCount = -1
	for i in range(0, lastExitTime + 1):
		current += vehicleCount[i]
		if(maxVehicleCount < current):
			maxVehicleCount = current
			index = i
	return maxVehicleCount

a = int(input())
start = end = []
for x in range(a):
	startTime, endTime = map(int, input().split())
	start.append(startTime)
	end.append(endTime)

time = peakTime(start, end)
print(time)

File: Uber/3/3.cpp
#include<iostream>
#include<vector>

int peakTime(std::vector<int> start, std::vector<int> end) {
	if (start.size() == 0) {
		return -1;
	}
	int lastExitTime = 0;
	for (int i = 0; i < start.size(); i++) {
		lastExitTime = std::max(lastExitTime, start[i]);
		lastExitTime = std::max(lastExitTime, end[i]);
	}
	int vehicleCount[lastExitTime + 2] = {0};
	int current = 0;
	int index = 0;
	for (int i = 0; i < start.size(); i++) {
		vehicleCount[start[i]] += 1;
		vehicleCount[end[i] + 1] -= 1;
	}
	int maxVehicleCount = -1;
	for (int i = 0; i <= lastExitTime; i++) {
		current += vehicleCount[i];
		if (maxVehicleCount < current) {
			maxVehicleCount = current;
			index = i;
		}
	}
	return maxVehicleCount;
}

int main() {
	int n;
	std::cin >> n;
	std::vector<int> start, end;
	while (n--) {
		int startTime, endTime;
		std::cin >>  startTime >> endTime;
		start.push_back(startTime);
		end.push_back(endTime);
	}
	std::cout << peakTime(start, end) << "\n";
	return 0;
}
